test/udp_sender.js

```diff
@@ -249,4 +250,20 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', () => {
+        sender._client.on('error', err => {
+            assert.equal(err.message, 'getaddrinfo ENOTFOUND foo.bar.com')
```

Comment:
I am  not saying that the event is not emitted, rather that the test is not testing the actual event handler. Can you use sinon to mock console.log? If so, you don't have to add another event in the test, and you will validate that the event handler in the sender is actually invoked, by validating that console.log was invoked.